runtime/metricsview/astexpr.go

```diff
@@ -210,7 +224,12 @@ func (b *sqlExprBuilder) writeBinaryCondition(exprs []*Expression, op Operator)
 
 		// If not unnested, write the expression as-is
 		if !unnest {
-			return b.writeBinaryConditionInner(nil, right, leftExpr, op)
+			if lm, ok := b.lookupMeta[left.Name]; ok {
+				leftExpr = fmt.Sprintf("%s IN (SELECT %s FROM dictionary(%s) WHERE %s ", b.ast.dialect.EscapeIdentifier(lm.keyExpr), b.ast.dialect.EscapeIdentifier(lm.keyCol), b.ast.dialect.EscapeIdentifier(lm.table), b.ast.dialect.EscapeIdentifier(lm.valueCol))
```

Comment:
Since this is ClickHouse-specific syntax, it should be implemented generically in a function on `drivers.Dialect`